### PR TITLE
Add zookeeper name in endpoint id

### PR DESCRIPTION
**Pull request by helifu**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added `enable_the_endpoint_id_with_zookeeper_name_prefix` setting for `ReplicatedMergeTree` (disabled by default). 
When enabled, it adds ZooKeeper cluster name to table's interserver communication endpoint. It avoids `Duplicate interserver IO endpoint` errors when having replicated tables with the same path, but different auxiliary ZooKeepers. 

When we migrate a replicated table from one zookeeper cluster to another (the reason why we migration is that zookeeper's load is too high), we will create a new table with the same zpath, but it will fail and the old table will be in trouble.

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
This is an automated comment for commit 9deb75b249619b7abdd38e3949ca8b3a76c9df8e with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49780/9deb75b249619b7abdd38e3949ca8b3a76c9df8e/ci_running.html)
The overall status of the commit is 🔴 failure

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🔴 failure</td></tr>
</table>

**Comment by tavplubix** [comment]
This change is backward incompatible and probably that's why an integration test `test_backward_compatibility/test.py::test_backward_compatability1` fails. 

Also, there's a similar PR: https://github.com/ClickHouse/ClickHouse/pull/49770

**Comment by helifu** [comment]
@tavplubix thanks for your info. Indeed, it's a backward incompatible problem which I noticed two days ago. Emm, maybe we could add a switch to the new endpoint id, could we? The default behavior is still the old design, but if users want this new improvement they can upgrade all the nodes and turn on the switch. What's your opinion?

**Comment by tavplubix** [comment]
Yes, seems like a setting is the only way to avoid incompatibility because incrementing `REPLICATION_PROTOCOL_VERSION` will not help in this case

**Review by tavplubix — APPROVED** [review]

**Comment by helifu** [comment]
@tavplubix 
Sorry, I have no idea to fix the failures, could you give me some tips?

**Comment by tavplubix** [comment]
Tests failures are not related to changes in this PR, Unit tests (asan) were broken in master, so it can be merged. However, I want to add a test from https://github.com/ClickHouse/ClickHouse/pull/49770, but it requires a small adjustment in the CI configuration.

**Comment by helifu** [comment]
> Tests failures are not related to changes in this PR, Unit tests (asan) were broken in master, so it can be merged. However, I want to add a test from #49770, but it requires a small adjustment in the CI configuration.

Awesome :) 👍

File: tests/queries/0_stateless/02442_auxiliary_zookeeper_endpoint_id.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
+-- Tags: no-fasttest
```
Comment (helifu) [inline tests/queries/0_stateless/02442_auxiliary_zookeeper_endpoint_id.sql:1]: Seems that the lastest failures have something to do with the long time the test case (02442_auxiliary_zookeeper_endpoint_id) takes.  Is there any way to shorten the test case run time?
Here is the error info I found:
`2023-05-24 18:02:28 02442_auxiliary_zookeeper_endpoint_id:                                  [ FAIL ] 236.52 sec. - Test runs too long (> 60s). Make it faster`

Comment (tavplubix) [reply tests/queries/0_stateless/02442_auxiliary_zookeeper_endpoint_id.sql]: This error can be ignored, and it's absolutely irrelevant to the `no-fasttest` tag, let's return the tag back.

**Comment by tavplubix** [comment]
Tests were OK on [6a30c08](https://github.com/ClickHouse/ClickHouse/pull/49780/commits/6a30c08d4476e61df4e349b24b0104563e7654ef), so let's merge

**Pull request merged**